waspc/data/Generator/templates/sdk/wasp/server/env.ts

```diff
@@ -0,0 +1,141 @@
+{{={= =}=}}
```

Comment:
All of env Wasp defined env variables live in this file - single central location that validates them. All env usage in the Wasp codebase should use this validated `env` object. 

Because we have different needs when in dev and production, we need to have different validations and default values (e.g. server URL is `http://localhost:3001` in dev by default but in production it is required!).

So, we have:
- `serverCommonSchema` (same for both dev and prod)
- `serverDevSchema` (used in dev, has defaults for some env vars)
- `serverProdSchema` (used in prod)

And then we have this bit of Zod "magic"
```ts
const serverEnvSchema = z.discriminatedUnion('NODE_ENV', [
  serverDevSchema.merge(serverCommonSchema),
  serverProdSchema.merge(serverCommonSchema)
])
```
which based on `NODE_ENV` picks one of the other schema to use. On type level it looks smth like this:
```ts
type Env =
  | {
      NODE_ENV: 'development'
      // optional in dev
      SOMETHING: string | undefined
    }
  | {
      NODE_ENV: 'production'
      // required in prod
      SOMETHING: string
    }
```